### DISCUSSION TITLE
Possible to get pypi downloads without mirrors?

### DISCUSSION THREAD

**Discussion by idnahacks** [badges/shields#7758] [Q&A] [answered]
Hi all,
I have a working badge for pypi downloads/month, but it includes the mirrors. I'm looking to be able to use the without mirrors query parameter to get a better understanding of how many real installs are occurring each month:
https://pypistats.org/api/packages/goodhound/overall

Is there a way I can do this with my shield?
https://img.shields.io/pypi/dm/goodhound

Apologies if I've missed something in the docs. I have searched.

**Comment by calebcartwright** [comment, marked as answer]
Something important to keep in mind is that Shields is really just focused on badges, and is reliant on the upstream systems of records to provide whatever data they have. In order to render our pypi downloads badge, we also fetch data from pypistats, and we use the number as they provide.

My understanding was that pypistats _does_ exclude mirrors by default. Whether or not they do, or whether they can but do so conditionally, is a question you would need to take up with them over in https://github.com/crflynn/pypistats.org

Let us know what you discover, as if it turns out the pypistats API does support conditionality via request parameters then we could certainly consider pulling those forward to our corresponding badge urls

**Comment by idnahacks** [comment]
Thanks for the response. I was mistaken in thinking that the number presented by pypistats included mirrors. Other pypistats api requests can have parameters to chose whether to include mirrors or not, but that's probably not a useful thing from a badge point of view.

It turns out that my package was just more popular than I thought! :D

Thanks again.
